src/pypowsybl_jupyter/selectcontext.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2024, RTE (http://www.rte-france.com)
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+#
+
+from collections import OrderedDict
+import pandas as pd
+from pypowsybl.network import Network
+
+class SelectContext:
+
+    def __init__(self, network:Network = None, vl_id : str = None, use_name:bool = True):
+        self.network = network
+        self.use_name = use_name
+
+        self.vls = network.get_voltage_levels(attributes=['name'])
+        self.vls['id'] = self.vls.index
+        if use_name:
+            self.vls['name'] = self.vls['name'].replace('', pd.NA).fillna(self.vls['id'])
+        else:
+            self.vls['name'] = self.vls.index
+        
+        self.vls = self.vls.sort_values(by='name')
+
+        self.apply_filter_by_name(None)
+
+        self.set_selected(self.vls.iloc[0].id if vl_id is None else vl_id)
+
+    def get_vls(self):
+        return self.vls
+    
+    def set_selected(self, id):
+        if id in self.vls.index:
+            self.selected_vl = id
+        else:
+            raise ValueError(f'a voltage level with id={id} does not exist in the network.')
+
+    def get_selected(self):
+        return self.selected_vl
+    
+    def apply_filter_by_name(self, sfilter):
```

Comment:
the naming is a bit problematic, "by_name" but it might be an id?

Reply:
This  context class is meant to support the selection list in the widget, a list of couples (name, id).  The 'name' is what is displayed in the selection list widget (could be not unique), then the id (assumed to be unique) is the actual id for the selection item to be used by this class clients.
This class builds this list of couples (name, id) according to the constructor parameter use_name (it tries also to fill the gaps where there are empty names or nulls) to use the network VL's names/ids.

So, the filter is always by_name but the name here is to be intended as the  selection list's 'name' elements, in the list of couples.